sktalk/corpus/conversation.py

```diff
@@ -1,11 +1,12 @@
 import warnings
+from typing import Optional
 from .utterance import Utterance
 from .write.writer import Writer
 
 
 class Conversation(Writer):
```

Comment:
A minor one: from a semantics and OOP perspective, I'm having a bit of a hard time imagining what/how a conversation can inherit from a writer. Is there any special reason that the base class is call writer?

Reply:
Thanks for this question, and I hope my answer makes sense, this is all quite new to me too!
`Writer` is not a proper parent class, but a class that is used to collect writer functionality that can be used by different objects. Is this weird architecture? Please let me know 😅 CC @carschno

Reply:
Thanks for the explanation @carschno! I didn't know about MixIns yet, but having a separate class that is used to provide some default methods here is logical to me. We indeed have another class that inherits from `Writer`: `Corpus`. There is some common functionality that we want to keep DRY as both `Conversation` and `Corpus` are objects that e.g. will be saved as json (#21) and csv (#47).